@xen-orchestra/rest-api/src/servers/server.controller.mts

```diff
@@ -0,0 +1,57 @@
+import { Example, Get, Path, Query, Request, Response, Route, Security, Tags } from 'tsoa'
+import { Request as ExRequest } from 'express'
+import { inject } from 'inversify'
+import { provide } from 'inversify-binding-decorators'
+import type { XoServer } from '@vates/types'
+
+import { notFoundResp, unauthorizedResp, type Unbrand } from '../open-api/common/response.common.mjs'
+import { partialServers, server, serverIds } from '../open-api/oa-examples/server.oa-example.mjs'
+import { RestApi } from '../rest-api/rest-api.mjs'
+import type { WithHref } from '../helpers/helper.type.mjs'
+import { XoController } from '../abstract-classes/xo-controller.mjs'
+
+@Route('servers')
+@Security('*')
+@Response(unauthorizedResp.status, unauthorizedResp.description)
+@Tags('servers')
+@provide(ServerController)
+export class ServerController extends XoController<XoServer> {
+  // --- abstract methods
+  _abstractGetObjects(): Promise<XoServer[]> {
+    return this.restApi.xoApp.getAllXenServers()
+  }
+  _abstractGetObject(id: XoServer['id']): Promise<XoServer> {
+    return this.restApi.xoApp.getXenServer(id)
+  }
+
+  constructor(@inject(RestApi) restApi: RestApi) {
+    super(restApi)
+  }
+
+  /**
+   * @example fields "status,id"
+   * @example filter "status:/^connected$/"
+   * @example limit 42
+   */
+  @Example(serverIds)
+  @Example(partialServers)
+  @Get('')
+  async getServers(
+    @Request() req: ExRequest,
+    @Query() fields?: string,
+    @Query() filter?: string,
+    @Query() limit?: number
+  ): Promise<string[] | WithHref<Unbrand<XoServer>>[] | WithHref<Partial<Unbrand<XoServer>>>[]> {
```

Comment:
is `WithHref<Unbrand<XoServer>>` needed since it already returns the partial ?  `WithHref<Partial<Unbrand<XoServer>>>[]`